File: problem-#1-by-Google/python/solution.py
def solution(list_nums, target):
    # use a set to cache the number we have seen.
    cache = set()
    for num in list_nums:
        if target - num in cache:
            return True
        else:
            cache.add(num)
    return False


def main():
    test_list = [10, 15, 3, 7]
    k = 17
    print(solution(test_list, k))


if __name__ == '__main__':
    main()

File: problem-#1-by-Google/javascript/solution.js
function solution(list_of_numbers, target) {
    var cache = new Set()
    for (let i = 0; i < list_of_numbers.length; i++) {
        if (cache.has(target - list_of_numbers[i])) {
            return true
        } 
        cache.add(list_of_numbers[i])
    }
    return false
}

var test_list = [10, 15, 3, 7]
var k = 17;

console.log(solution(test_list, k))